packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -151,27 +152,28 @@ export class AssociationManager {
     }
   }
 
-  public static associateTagOptions(portfolio: IPortfolio, tagOptions: TagOptions): void {
-    const portfolioStack = cdk.Stack.of(portfolio);
+
+  public static associateTagOptions(resource: IResource, resourceId: string, tagOptions: TagOptions): void {
```

Comment:
By doing it this way it minimizes necessary code changes.  The crux here was that we need the Id, which is a different field in both the portfolio and product.  I see two ways of doing this. one way would create two entry points for the logic here,  e.g. `AssociationManager` has two different methods for associating tag options, and those would then reuse the same underlying functions., or doing it this way.  I chose this way because it seemed more efficient, the code changes are just from renaming, and this method is only used internally so we control the types inputted.

Reply:
That or having just a type resolution method that can take some
``` ts
type productOrPortfolio = IPortfolio | IProduct;
```
and then resolving which one it is based on if it had the `portfolioId` field or not.  I think this type of approach was shunned before for a similar issue.

Reply:
Honestly, I don't think it matters too much 😜. The current solution is fine.

This is all module-private, we can change it any time we want.